src/lib/users/widgets/public_key_selector.rb

```diff
@@ -0,0 +1,282 @@
+        nil
+      end
+
+      # Forces this widget to listen to all events
+      #
+      # @return [Boolean]
+      def handle_all_events
+        true
+      end
+
+      # Determines whether a public key is selected or not
+      #
+      # @return [Boolean] true if empty; false otherwise
+      def empty?
+        value.nil?
+      end
+
+      # Helper method to get the current value (the selected public key)
+      #
+      # @return [SSHPublicKey,nil] Return the selected public key (nor nil if no key is selected)
+      def value
+        self.class.value
+      end
+
+      # Returns the help text regarding the use of a public key for authentication
+      #
+      # @return [String] Help text
+      def help
+        @help ||= _(
+          "<p>\n" \
+          "In some situations it is preferred to access to the system remotely via SSH\n" \
+          "using a public key instead of a password. This screen allows you to select\n" \
+          "one public key from an USB stick, a CD/DVD ROM or even from an existing\n" \
+          "partition.\n\n" \
+          "Take into account that the root password and the public key are not mutually\n" \
+          "exclusive: you can provide both if you want.\n" \
+          "</p>\n"
+        )
+      end
+
+    private
+
+      # Helper method to set the current value (the selected public key)
+      #
+      # @param [SSHPublicKey] Return the current public key
+      def value=(key)
+        self.class.value = key
+      end
+
+      def selected_blk_device_name
+        self.class.selected_blk_device_name
+      end
+
+      # Widget's inner content
+      #
+      # It displays the public key content or a disk selector if no key has been selected.
+      #
+      # @return [Yast::Term]
+      def inner_content
+        VBox(empty? ? blk_device_selector : public_key_content)
+      end
+
+      # Disk selector
+      #
+      # This widget displays includes a list of selectable disk and a button to browse the
+      # selected one.
+      #
+      # @return [Yast::Term]
+      def blk_device_selector
+        VBox(
+          Left(
+            HBox(
+              blk_devices_combo_box,
+              PushButton(Id(:refresh), Opt(:notify), _("Refresh..."))
+            )
+          ),
+          Left(PushButton(Id(:browse), Opt(:notify), _("Browse...")))
+        )
+      end
+
+      # UI which shows the public key content
+      #
+      # @return [Yast::Term]
+      def public_key_content
+        VBox(
+          Left(Label(value.fingerprint)),
+          HBox(
+            Left(Label(value.comment)),
+            Right(PushButton(Id(:remove), Opt(:notify), _("Remove")))
+          )
+        )
+      end
+
+      # Key comment to show to the user
+      #
+      # @note When no comment is present, the widget shows 'no comment' just as ssh-keygen does.
+      #
+      # @return [String]
+      def comment_value
+        value.comment || _("no comment")
```

Comment:
This will be hard to translate without the context, please add a `TRANSLATORS:` comment.